storage/series_test.go

```diff
@@ -322,13 +411,24 @@ func TestHistogramSeriesToChunks(t *testing.T) {
 
 func testHistogramsSeriesToChunks(t *testing.T, test histogramTest) {
 	lbs := labels.FromStrings("__name__", "up", "instance", "localhost:8080")
-	series := NewListSeries(lbs, test.samples)
+	copiedSamples := []tsdbutil.Sample{}
```

Comment:
@codesome I had to add this copy for gauge histograms as the histogram was modified in place (spans and buckets, nothing else). Is that expected?

Reply:
Copied in `NewListSeries` or somewhere else?

Reply:
I think the modification is happening in the `seriesToChunkEncoder` when there is recoding. The recoding also happens in TSDB append, and if it mutating the histogram, it is likely breaking the interface as described [here](https://github.com/prometheus/prometheus/blob/7c2de14b0bd74303c2ca6f932b71d4585a29ca75/storage/interface.go#L282-L284). I will open a separate conversation on that.

Reply:
This is coming from https://github.com/prometheus/prometheus/blob/7c2de14b0bd74303c2ca6f932b71d4585a29ca75/storage/series.go#L482  Which has "Recoding" in the name, and calls RecordHistogram which does the change in place. So it's kind of understandable...